pyccl/tests/test__caching.py

```diff
@@ -147,5 +147,13 @@ def func2():
         ccl.Caching.policy = "my_policy"
 
 
+def test_caching_parentheses():
+    """Verify that ``cache`` can be used with or without parentheses."""
+    func1, func2 = [lambda: None for _ in range(2)]
+    ccl.cache(func1)
+    ccl.cache(maxsize=10, policy="lru")(func2)
+    assert all([hasattr(func, "cache_info") for func in [func1, func2]])
+
+
```

Comment:
why this new test?

Reply:
covers the red line here, nothing to do with this PR
![Screenshot from 2023-04-01 05-03-00](https://user-images.githubusercontent.com/22447385/229260642-0ef0c800-6609-4378-b5a0-1c4c5cb43468.png)